src/main/blackbox/blackbox.c

```diff
@@ -1181,7 +1200,7 @@ static void loadMainState(timeUs_t currentTimeUs)
     blackboxMainState_t *blackboxCurrent = blackboxHistory[0];
 
     blackboxCurrent->time = currentTimeUs;
-
+    float imuAttitude[] = {imuAttitudeQuaternion.x, imuAttitudeQuaternion.y, imuAttitudeQuaternion.z};
```

Comment:
It is easier to update quaternion definition to match `vector3_t`:
```
typedef union {
    float v[4];
    struct {
        float w,x,y,z;
    };
} quaternion;
```

And maybe rename it to `quaternion_t`, it is used only in few places, so it may fit this PR ...

Reply:
But take care, first element is W, so you must use `i+1` with comment

Maybe this helper variable is better, just make it `const`

Reply:
[Checked on my tiny whoop](https://github.com/betaflight/blackbox-log-viewer/pull/790). 
Looks good.
The quaternion code refactoring in process.